deploy/webhook.yaml

```diff
@@ -0,0 +1,258 @@
+    resources:
+      - validatingwebhookconfigurations
+    verbs:
+      - get
+      - update
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: service-api-admission
+  annotations:
+  labels:
+    name: service-api-webhook
+    version: 0.0.1
+  namespace: service-api
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: service-api-admission
+subjects:
+  - kind: ServiceAccount
+    name: service-api-admission
+    namespace: service-api
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: service-api-admission
+  annotations:
+  labels:
+    name: service-api-webhook
+    version: 0.0.1
+  namespace: service-api
+rules:
+  - apiGroups:
+      - ''
+    resources:
+      - secrets
+    verbs:
+      - get
+      - create
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: service-api-admission
+  annotations:
+  labels:
+    name: service-api-webhook
+    version: 0.0.1
+  namespace: service-api
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: service-api-admission
+subjects:
+  - kind: ServiceAccount
+    name: service-api-admission
+    namespace: service-api
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    name: service-api-webhook
+    version: 0.0.1
+  name: service-api-admission
+  namespace: service-api
+spec:
+  type: ClusterIP
+  clusterIP: 10.96.0.2
+  ports:
+    - name: https-webhook
+      port: 443
+      targetPort: 8443
+  selector:
+    name: service-api-admission
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: service-api-admission
+  namespace: service-api
+  labels:
+    name: service-api-admission
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: service-api-admission
+  template:
+    metadata:
+      name: service-api-admission
+      labels:
+        name: service-api-admission
+    spec:
+      containers:
+        - name: webhook
+          image: cmluciano/service-api-admission:latest
+          imagePullPolicy: Always
+          args:
+            - -alsologtostderr
+            - --log_dir=/
+            - --tlsCertFile=/etc/certs/cert
+            - --tlsKeyFile=/etc/certs/key
+            - -v=10
+            - 2>&1
+          ports:
+            - containerPort: 8443
+              name: webhook
+          resources:
+            limits:
+              memory: 50Mi
+              cpu: 300m
+            requests:
+              memory: 50Mi
+              cpu: 300m
+          volumeMounts:
+            - name: webhook-certs
+              mountPath: /etc/certs
+              readOnly: true
+            - name: logs
+              mountPath: /tmp
+          securityContext:
+            readOnlyRootFilesystem: true
+      volumes:
+        - name: webhook-certs
+          secret:
+            secretName: service-api-admission
+        - name: logs
+          emptyDir: {}
+---
+apiVersion: batch/v1
```

Comment:
That makes sense to me, but I worry it will cause some strange overhead for straightforward testing without something like helm. I will put a TODO here to make this a helm based deploy and ensure that all of these can be skipped from the helm config if requested.

Reply:
That makes sense, I will do that now.

Reply:
Can we also please add a single line doc line to each of these setup jobs to explain what they are doing?